client/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -2,36 +2,55 @@ package org.http4s
 package client
 package middleware
 
-import org.http4s.Status.ResponseClass.Successful
-import org.http4s.{Response, Request, EmptyBody}
-import org.http4s.client.Client
-import org.log4s.getLogger
-
 import scala.concurrent.duration._
 import scala.math.{pow, min, random}
-import scalaz.Kleisli
 
+import scalaz._
 import scalaz.concurrent.Task
+import org.http4s.Status._
+import org.log4s.getLogger
 
 object Retry {
 
   private[this] val logger = getLogger
 
+  private[this] val RetriableStatuses = Set(
+    RequestTimeout,
+    // TODO Leaving PayloadTooLarge out until we model Retry-After
+    InternalServerError,
+    ServiceUnavailable,
+    BadGateway,
+    GatewayTimeout
+  )
```

Comment:
RequestTimeout, PayloadTooLarge, and ServiceUnavailable are explicitly permitted retries by the spec.  The others feel right to me.

Does anybody have a use case for customizing this further?  To do so would complicate the signature.